### DISCUSSION TITLE
Opacity for NaN-values when adding mesh

### DISCUSSION THREAD

**Discussion by georg-walther** [pyvista/pyvista#3498] [Q&A] [answered]
Dear PyVista-Team,
Dear PyVista-Community,

is there any workaround to make NaN values invisible (opacity=0) for a mesh? In [this issue](https://github.com/pyvista/pyvista/issues/3445) I reported it as a bug, but any workaround for this would also be fine. Examples to reproduce are also included in the above link. I tried several things without success. 

Many thanks in advance

**Comment by banesullivan** [comment, marked as answer]
I like to run a [`threshold()`](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSetFilters.threshold.html) filter to remove nans. The default for that filter if no arguments are given is to remove the nans. Here is a modified version of your example:

```py
import pyvista as pv
import numpy as np

noise = pv.perlin_noise(0.1, (1, 1, 1), (0, 0, 0))
grid = pv.sample_function(noise, [0, 1.0, -0, 1.0, 0, 1.0], dim=(20, 20, 20))

minV = min(grid['scalars'])
maxV = max(grid['scalars'])
grid['scalars'] = (grid['scalars'] - minV) / (maxV - minV)
grid['scalars'] *= 100

grid['scalars'] = np.where(grid['scalars'] < 20, np.nan, grid['scalars'])

no_nan = grid.threshold(scalars='scalars')

p = pv.Plotter(notebook=0)
p.add_mesh(no_nan)
p.show()
```

<img width="624" alt="Screen Shot 2022-10-21 at 9 37 53 AM" src="https://user-images.githubusercontent.com/22067021/197234595-e410f109-b993-41c7-a808-34d27924d5d6.png">

However, you can see the boundary of those nan values still showing up...

I've discussed why this happens in https://github.com/pyvista/pyvista/discussions/1758#discussioncomment-1671816

To work around this, and leverage the fact that the threshold filter is a cell-wise operation, you can run the [`point_data_to_cell_data()`](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSetFilters.point_data_to_cell_data.html) filter:

```py
# ptc is an alias for point_data_to_cell_data
no_nan = grid.ptc().threshold(scalars='scalars')

p = pv.Plotter(notebook=0)
p.add_mesh(no_nan)
p.show()
```
<img width="624" alt="Screen Shot 2022-10-21 at 9 41 09 AM" src="https://user-images.githubusercontent.com/22067021/197235323-37fd7aa7-f32b-40be-9f3d-ff025ad049ce.png">
